### DISCUSSION TITLE
Additional joistick mode, off?

### DISCUSSION THREAD

**Discussion by tuFFo126** [UzixLS/zx-sizif-512#55] [Ideas]
Joistick modes, would it be possible to add an off option in the menu?  It's like when some external controllers don't have the option to turn them off.  Usually it is Kempston, in fact, to avoid a conflict, you can choose the second one, but sometimes it is better to turn it off, it would be useful.

**Comment by UzixLS** [comment]
In what cases would such an option be useful?

**Reply by tuFFo126** [reply]
I have an external k-mous turbo interface that allows me to set kempston, sinclair, qaop keymapping and of course kempston mouse modes.  Then you need to remember to turn on the joistick in the Sizif, which may not duplicate.  Some interfaces force you to work with the default configuration (even if you save your initial settings), then there is the Kempston joystick, the Czech mini beta disk 128 has a conflict with Kempston and forces the default configuration, but then loading the begining setting. off will not help anyway.

**Reply by UzixLS** [reply]
What I think should be done to improve compatibility with these addons:

1. Detect addon with kempston joystick at startup by probing #DF (or #1F? or both?) port. If it's value != 0xFF - force Sizif's joystick to Sinclair mode. 
A similar algorithm is already implemented for AY, TurboSound and BDI.

2. Make Sinclair mode default in cases when saved settings cannot be loaded (when Sizif's ROM is blocked by addon).

**Reply by tuFFo126** [reply]
Is setting default mode different than saving startup settings?  When I saved the settings on the Sinclair joistick, entering the interfaces that block the sizifa menu, it turns out that the Kempston joistick is active anyway.

**Reply by UzixLS** [reply]
There is volatile registers in FPGA that stores actual configuration. They internally sets to default values on power-on and (this will be eliminated in next firmware version) on reset. After that, on power-on Sizif's ROM starts and reconfigure these registers using configuration saved by user. This ROM can be blocked by addon with ROMCSB signal. In this case user configuration will not be loaded.

**Reply by tuFFo126** [reply]
So in this case setting and saving as a Sinclair joystick won't do anything?

**Reply by UzixLS** [reply]
If the ROM is blocked, settings will be set to default values, and the menu is unavailable.

**Reply by UzixLS** [reply]
Please try this firmware version. In addition to the above changes, it includes possible compatibility fix for some BDI addons, can you check this also?

This is a development version, and unfortunately there may be issues with some SRAM chips.

[sizif512_20240128.zip](https://github.com/UzixLS/zx-sizif-512/files/14077017/sizif512_20240128.zip)

**Reply by tuFFo126** [reply]
I don't have such opportunities, but I will tell my friend what sizify does.  Have you checked these false ULA+ colors, what could be the reason?

**Reply by UzixLS** [reply]
@tuFFo126 in TODO list

**Reply by tuFFo126** [reply]
It's nice and I'm happy.  In overcolored games this is not a problem, but in images false colors are a problem.
